### ISSUE TITLE
443: bind: permission denied

### ISSUE THREAD

**Issue by lemos1235** [caddyserver/caddy-docker#21]
I had a docker-compose file as below: 
```
 caddy:
    image: caddy/caddy:alpine
    #restart: always
    ports:
      - '80:80'
      - '443:443'
    volumes:
      - ./Caddyfile:/etc/caddy/Caddyfile
    environment:
      - ACME_AGREE=true
```

but got the error log:
loading initial config: loading new config: http app module: start: tcp: listening on :443: listen tcp :443: bind: permission denied

**Comment by Ilyes512** [comment]
You need to use a higher port number in your Caddyfile (>1024).  You can make it work with a port lower then 1024, but you will need some more steps like allowing the container to bind to service ports (--cap-add NET_BIND_SERVICE) and give the actual service the right to do this as an unprivileged user. 

I recommend you use 8080 and 4430 and bind it like:

```
ports:
  - "80:8080"
  - "443:4430"
```

**Comment by hairyhenderson** [comment]
Fixed by #30 - the image now runs as root by default, so `80`/`443` will work fine.
